jax/numpy/lax_numpy.py

```diff
@@ -3316,7 +3321,7 @@ def _index_to_gather(x_shape, idx):
   collapsed_slice_dims = []
   start_index_map = []
 
-  index_dtype = int64 if _max(x_shape, default=0) >= (1 << 31) else int32
+  index_dtype = int64 if any([type(dim) is Poly or dim >= (1 << 31) for dim in x_shape]) else int32
```

Comment:
Btw we need to use `_any` not `any` here, since the latter is a `jax.numpy` function! Tricky.